File: Source/iOS/Genealogy/Genealogy/Model/COM/DTO/UserInfoDTO.m
//
//  UserInfoDTO.m
//  Genealogy
//
//  Created by xiao huama on 15/09/10.
//  Xiao Hua Ma personal studio 2015.
//

#import "UserInfoDTO.h"

@implementation UserInfoDTO

- (void)loadDataWithJsonData:(id)jsonData
{
    if (![jsonData isKindOfClass:[NSDictionary class]]) {
        return;
    }
    if (!jsonData) {
        return;
    }
    if (![jsonData[@"userID"] isKindOfClass:[NSNull class]]) {
        self.userID = jsonData[@"userID"];
    }
    if (![jsonData[@"familyCode"] isKindOfClass:[NSNull class]]) {
        self.familyCode = jsonData[@"familyCode"];
    }
    if (![jsonData[@"name"] isKindOfClass:[NSNull class]]) {
        self.name = jsonData[@"name"];
    }
    if (![jsonData[@"sex"] isKindOfClass:[NSNull class]]) {
        self.sex = jsonData[@"sex"];
    }
    if (![jsonData[@"birthday"] isKindOfClass:[NSNull class]]) {
        self.birthday = jsonData[@"birthday"];
    }
    if (![jsonData[@"mobile"] isKindOfClass:[NSNull class]]) {
        self.mobile = jsonData[@"mobile"];
    }
    if (![jsonData[@"maile"] isKindOfClass:[NSNull class]]) {
        self.maile = jsonData[@"maile"];
    }
    if (![jsonData[@"backPath"] isKindOfClass:[NSNull class]]) {
        self.backPath = jsonData[@"backPath"];
    }
    if (![jsonData[@"selfPath"] isKindOfClass:[NSNull class]]) {
        self.selfPath = jsonData[@"selfPath"];
    }
    if ([jsonData[@"picture"] isKindOfClass:[NSArray class]]) {
        if ([jsonData[@"picture"] count]>0) {
            NSMutableArray *tempArray = [NSMutableArray array];
            for (id data in jsonData[@"picture"]) {
                [tempArray addObject:data];
            }
            self.picture = tempArray;
        }
    }
    if ([jsonData[@"video"] isKindOfClass:[NSArray class]]) {
        if ([jsonData[@"video"] count]>0) {
            NSMutableArray *tempArray = [NSMutableArray array];
            for (id data in jsonData[@"video"]) {
                [tempArray addObject:data];
            }
            self.video = tempArray;
        }
    }
    if (![jsonData[@"profile"] isKindOfClass:[NSNull class]]) {
        self.profile = jsonData[@"profile"];
    }
    if ([jsonData[@"address"] isKindOfClass:[NSDictionary class]]) {
        self.address = [[AddressInfoDTO alloc] init];
        [self.address loadDataWithJsonData:jsonData[@"address"]];
    }
}

-(NSMutableDictionary *)buildRequestData
{
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    if (self.userID) {
        dic[@"userID"] = self.userID;
    }
    if (self.familyCode) {
        dic[@"familyCode"] = self.familyCode;
    }
    if (self.name) {
        dic[@"name"] = self.name;
    }
    if (self.sex) {
        dic[@"sex"] = self.sex;
    }
    if (self.birthday) {
        dic[@"birthday"] = self.birthday;
    }
    if (self.mobile) {
        dic[@"mobile"] = self.mobile;
    }
    if (self.maile) {
        dic[@"maile"] = self.maile;
    }
    if (self.backPath) {
        dic[@"backPath"] = self.backPath;
    }
    if (self.selfPath) {
        dic[@"selfPath"] = self.selfPath;
    }
    if (self.profile) {
        dic[@"profile"] = self.profile;
    }
    if (self.picture.count>0) {
        dic[@"picture"] = self.picture;
    }
    if (self.video.count>0) {
        dic[@"video"] = self.video;
    }
    NSDictionary *addressDic = [self.address buildRequestData];
    if (addressDic) {
        dic[@"address"] = addressDic;
    }
    return dic;
}

@end

File: Source/Service/appService/com.maxiaohua.genealogy.main.biz/src/com/maxiaohua/genealogy/main/biz/com/dto/UserInfoDTO.java
﻿/*
 * Genealogy
 * Xiao Hua Ma personal studio 2015.
 */
package com.maxiaohua.genealogy.main.biz.com.dto;

/**
 * UserInfoDTO DTO
 *
 * History
 * REV.         Updated Date           Updater              Infomation
 * -------      ---------------        ----------------     ------------------
 * 1.0          2015/09/06             TOOL                Create
*
 */
@com.maxiaohua.genealogy.fw.core.type.AliasKanJi("用户信息")
public class UserInfoDTO implements java.io.Serializable, Comparable<UserInfoDTO> {
	/**
	* System ID
	*/
	private static final long serialVersionUID = 1L;

	private String userID = null;

	public String getUserID() {
		return this.userID;
	}

	public void setUserID(
			String userID) {
		this.userID = userID;
	}

	private String familyCode = null;

	public String getFamilyCode() {
		return this.familyCode;
	}

	public void setFamilyCode(
			String familyCode) {
		this.familyCode = familyCode;
	}

	private String name = null;

	public String getName() {
		return this.name;
	}

	public void setName(
			String name) {
		this.name = name;
	}

	private Integer sex = null;

	public Integer getSex() {
		return this.sex;
	}

	public void setSex(
			Integer sex) {
		this.sex = sex;
	}

	private java.sql.Date birthday = null;

	public java.sql.Date getBirthday() {
		return this.birthday;
	}

	public void setBirthday(
			java.sql.Date birthday) {
		this.birthday = birthday;
	}

	private String mobile = null;

	public String getMobile() {
		return this.mobile;
	}

	public void setMobile(
			String mobile) {
		this.mobile = mobile;
	}

	private String maile = null;

	public String getMaile() {
		return this.maile;
	}

	public void setMaile(
			String maile) {
		this.maile = maile;
	}

	private String backPath = null;

	public String getBackPath() {
		return this.backPath;
	}

	public void setBackPath(
			String backPath) {
		this.backPath = backPath;
	}

	private String selfPath = null;

	public String getSelfPath() {
		return this.selfPath;
	}

	public void setSelfPath(
			String selfPath) {
		this.selfPath = selfPath;
	}

	private String[] picture = null;

	public String[] getPicture() {
		return this.picture;
	}

	public void setPicture(
			String[] picture) {
		this.picture = picture;
	}

	private String[] video = null;

	public String[] getVideo() {
		return this.video;
	}

	public void setVideo(
			String[] video) {
		this.video = video;
	}

	private String profile = null;

	public String getProfile() {
		return this.profile;
	}

	public void setProfile(
			String profile) {
		this.profile = profile;
	}

	private AddressInfoDTO address = null;

	public AddressInfoDTO getAddress() {
		return this.address;
	}

	public void setAddress(
			AddressInfoDTO address) {
		this.address = address;
	}

  	/* (non-Javadoc)
     * @see java.lang.Comparable#compareTo(java.lang.Object)
     */
    @Override
    public int compareTo(
    		UserInfoDTO compareToDTO) {
        return this.equals(compareToDTO) ? 0 : -1;
    }
}
